IRSA/terraform/README.md

```diff
@@ -7,60 +7,175 @@ across (or within) AWS accounts.
+            ],
+            "Sid": "IRSAAdmin"
+        }
+    ],
+    "Version": "2012-10-17"
+}
+```
+
+The `assume_role_policy` for this role ties to the IRSA service account within the cluster in question:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "<EKS_CLUSTER_OIDC_PROVIDER_ARN>"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "<EKS_CLUSTER_OIDC_PROVIDER_HOSTNAME>:sub": "system:serviceaccount:domino-field:irsa"
+                }
+            }
+        }
+    ]
+}
+```
+
+You should have 3x proxy roles as well: `demo-domino-proxy-list-bucket-role-1`, `demo-domino-proxy-read-bucket-role-1`, `demo-domino-proxy-write-bucket-role-1`. Each of these roles will have a policy similar to this:
+
+```json
+{
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Effect": "Allow",
+            "Resource": <ASSOCIATED_WORKLOAD_ROLE_ARN>
+        }
+    ],
+    "Version": "2012-10-17"
+}
+```
+
+where <ASSOCIATED_WORKLOAD_ROLE_ARN> is the ARN of the workload role associated to the proxy role. **NOTE** proxy roles have a many:1 association with workload roles. Each workload role may be associated with several proxy roles, but each proxy role will only be associated with a single workload role.
+
+Each proxy role will also have an `assume_role_policy` similar to this:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "<EKS_CLUSTER_OIDC_PROVIDER_ARN>"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "<EKS_CLUSTER_OIDC_PROVIDER_HOSTNAME>:sub": ""
+                }
+            }
+        }
+    ]
+}
+```
+
+Note that the `sub` value here is empty: this will be modified as needed by the IRSA service to add/remove the service account(s) for a given run, based on the contents of the `ConfigMap` "domino-org-iamrole-mapping".
+
+### Account: **asset-acct**
+
+This account is where all of the workload IAM roles will be found. With this codebase, there are three: `demo-domino-list-bucket-role`, `demo-domino-read-bucket-role`, `demo-domino-write-bucket-role`. Since this is a demo codebase, we only have one policy for all three workloads (despite the naming differences to the contrary). All of the workload roles in this case use the list-bucket policy defined in [demo_workload_policy.tf](demo_workload_policy.tf):
+
+```json
+{
+    "Statement": [
+        {
+            "Action": "s3:ListBucket",
+            "Effect": "Allow",
+            "Resource": "*"
+        }
+    ],
+    "Version": "2012-10-17"
+}
+```
+
+Each workload policy also has an `assume_role_policy` similar to this:
```

Comment:
Should be the other way around right? A proxy role assumes the workload role. And the workload role has a trust policy which allows the proxy role to assume the workload role.